### ISSUE TITLE
Feature Req: Change job name before octoprint upload

### ISSUE THREAD

**Issue by soostdijck** [rgomezjnr/GcodeFilenameFormatPlus#1]
It would be great if there was a way to give the job a custom name before uploading to octoprint. Sort of like a "save as" feature. Especially with multiple objects in a single print, the default objectname isn't great. 

This doesn't apply to saving local files, because that gives you a filename dialog before actually saving the file to disk.

p.s. I'm not sure if this feature would be at home in this plugin, or in the octoprint plugin itsself.

**Comment by soostdijck** [comment]
Okay I just noticed the naming dialog in the bottom left or cura, oof! Closing

**Comment by Jordan87** [comment]
Yes, but if i will modify name there then plugin is not working and not adding configured parameters. Its working only if i will not modify name there. :/
Cura 4.12.1
GcodeFilenameFormatPlus 0.1.0

Thanks

**Comment by rgomezjnr** [comment]
> p.s. I'm not sure if this feature would be at home in this plugin, or in the octoprint plugin itsself.

@soostdijck Addressing this in GFF+ is the correct place since GFF+ modifies the job name before it is sent to OctoPrint.

> Especially with multiple objects in a single print, the default objectname isn't great.

The [object_count] option may help you with a more appropriate filename format.

> Yes, but if i will modify name there then plugin is not working and not adding configured parameters. Its working only if i will not modify name there. :/

@Jordan87 The behavior I'm seeing is that modifying the name (using the field under the Object list in the lower left-hand corner) before slicing does work, but only once. Modifying the name after that, without re-slicing, and simply selecting the Save to Disk button does not work, the filename format is not used. Can you replicate this behavior? I tested this with the same versions as you:

Cura 4.12.1
GcodeFilenameFormatPlus 0.1.0

First name edit:

![Screenshot 2021-12-15 184006](https://user-images.githubusercontent.com/55524335/146300138-e319ca1b-a471-4cf2-a9ab-37e4e70d61ef.png)

Second name edit:

![Screenshot 2021-12-15 184057](https://user-images.githubusercontent.com/55524335/146300084-100ab309-d080-45a3-83e7-bc4ffe6e3535.png)

**Comment by Jordan87** [comment]
Yes, you are corrrect. Its working only if i change name before slicing :)

**Comment by rgomezjnr** [comment]
I'll see if I can get the filename format to work after editing the name.

**Comment by rgomezjnr** [comment]
I haven't worked on this issue yet. I'll ask if other developers can help with this.
